backend/src/main/java/eventlistener/service/EventService.java

```diff
@@ -0,0 +1,66 @@
+package eventlistener.service;
+
+import eventlistener.model.notificationUser.NotificationUser;
+import eventlistener.model.event.Event;
+import eventlistener.repo.EventRepo;
+import eventlistener.repo.NotificationUserRepo;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+
+@Service
+public class EventService {
+
+    private final EventRepo eventRepo;
+
+
+    public EventService(EventRepo eventRepo) {
+        this.eventRepo = eventRepo;
+
+    }
+
+    public List<Event> getAllEvents() {
+        return eventRepo.findAll();
+    }
+
+    public Event addEvent(Event eventToAdd) {
+        return eventRepo.save(eventToAdd);
+    }
+
+    public List<Event> getAllEventsFromUser(String userId) {
+        List<Event> userEvents = new ArrayList<>();
+        List<Event> allEvents = eventRepo.findAll();
+        for (Event event : allEvents) {
+            List<String> users = event.getNotificationUser();
+            for (String user : users) {
+                if(Objects.equals(user, userId)){
+                    userEvents.add(event);
+                }
+            }
+        }
+        return userEvents;
+    }
```

Comment:
Within your `EventRepo` you can add the following query:

`return eventRepo.findAllByNotificationUserContains(userId);`

```suggestion
    public List<Event> getAllEventsFromUser(String userId) {
return eventRepo.findAllByNotificationUserContains(userId);
    }
```